pkg/db/interface.go

```diff
@@ -1,9 +1,27 @@
 package db
 
-type ID = string
+type FileID = string
+
+type CategoryID = string
+type DocumentText struct {
+	DocumentID FileID
+	Text       string
+}
+type Category struct {
+	ID    CategoryID     `json:"id"`
+	Name  string         `json:"name"`
+	Texts []DocumentText `json:"texts"`
+}
+type Categories struct {
+	Categories map[string]Category `json:"categories"`
+}
 
 type Store interface {
-	UploadFile(filename string, contents []byte) (ID, error)
-	GetFile(id ID) ([]byte, error)
-	Files() ([]ID, error)
+	UploadFile(filename string, contents []byte) (FileID, error)
+	GetFile(id FileID) ([]byte, error)
+	Files() ([]FileID, error)
+	CreateCategory(name string) (Category, error)
```

Comment:
Sorry, I think I screwed this up last time. Let's just have CreateCategory return a CategoryID, the category will be empty anyways, and invalid as soon as the underlying category is changed. On the other hand the ID stays valid.